test/context_test.go

```diff
@@ -136,6 +137,24 @@ func testContextRequestWithTimeoutCanceled(t *testing.T, nc *nats.Conn) {
 	// Cancel the context already so that rest of requests fail.
 	cancelCB()
 
+	// Wait a bit for context to be eventually canceled.
+	errCh := make(chan error)
+	tf := time.AfterFunc(50*time.Millisecond, func() {
```

Comment:
Sometimes can't avoid sleeps but where possible I like for this to be more deterministic. I added a waitFor(condition, timeout) helper function in my drain branch.

Reply:
great, there might be a few staticcheck related conflicts with the drain branch so thinking on branching from it and already adopt the new helper instead

Reply:
ok I can merge your changes into the drain branch if you want. If so just cancel this PR.